<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>


  <node pkg="stage_ros" type="stageros" name="stageros" args="$(find navigation_empty_map)/world/square20m.world" respawn="false">
    <param name="base_watchdog_timeout" value="0.2"/>
  </node>

<!--
  <node pkg="map_server" type="map_server" name="map_server" args="$(find navigation_empty_map)/world/square20m.yaml" respawn="false" >
    <param name="frame_id" value="/map" />
  </node>
-->

  <include file="$(find navigation_empty_map)/launch/includes/gmapping.launch.xml">
    <arg name="scan_topic" value="robot_0/base_scan" />
  </include>


  <!-- BEGIN ROBOT 0 -->
  <group ns="robot_0">
    <param name="tf_prefix" value="robot_0" />

    <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
      <remap from="map" to="/map" />
      <param name="controller_frequency" value="10.0" />
      <rosparam file="$(find navigation_empty_map)/param_sim/costmap_common_params.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find navigation_empty_map)/param_sim/costmap_common_params.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find navigation_empty_map)/param_sim/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find navigation_empty_map)/param_sim/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find navigation_empty_map)/param_sim/base_local_planner_params.yaml" command="load" />
    </node>

    <node pkg="fake_localization" type="fake_localization" name="fake_localization" respawn="false" output="screen">
      <param name="odom_frame_id" value="robot_0/odom" />
      <param name="base_frame_id" value="robot_0/base_footprint" />
    </node>

  </group>
  <!-- END ROBOT 0 -->


  <node name="robot_0" pkg="navigation_empty_map" type="robot_move" output="screen" >
    <param name="tf_prefix" value="robot_0" />
  </node>


  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find navigation_empty_map)/one_robot.rviz" />


</launch>
